linux/software/libcamera/csi-2-usage.md

```diff
@@ -0,0 +1,168 @@
+
+As with analogue bridges, the timings typically fix the V4L2 CAPTURE queue resolution, and calling `VIDIOC_S_DV_TIMINGS` with the result of `VIDIOC_QUERY_DV_TIMINGS` before streaming should ensure the format is correct.
+
+Depending on the bridge chip and the driver, it may be possible for changes in the input source to be reported to the application via `VIDIOC_SUBSCRIBE_EVENT` and `V4L2_EVENT_SOURCE_CHANGE`.
+
+#### Currently supported devices
+
+There are 2 bridge chips that are currently supported by the Rasberry Pi Linux kernel, the Analog Devices ADV728x-M for analogue video sources, and the Toshiba TC358743 for HDMI sources. 
+
+
+*Analog Devices ADV728x(A)-M Analogue video to CSI2 bridge*
+
+These chips convert composite, S-video (Y/C), or component (YPrPb) video into a single lane CSI-2 interface, and are supported by the [ADV7180 kernel driver](https://github.com/raspberrypi/linux/blob/rpi-5.4.y/drivers/media/i2c/adv7180.c).
+
+Product details for the various versions of this chip can be found on the Analog Devices website. 
+
+[ADV7280A](https://www.analog.com/en/products/adv7280a.html), [ADV7281A](https://www.analog.com/en/products/adv7281a.html), [ADV7282A](https://www.analog.com/en/products/adv7282a.html)
+
+Because of some missing code in the current core V4L2 implementation, selecting the source fails, so the Raspberry Pi kerrnel version adds a kernel module parameter called `dbg_input` to the ADV7180 kernel driver which sets the input source every time VIDIOC_S_STD is called. At some point mainstream will fix the underlying issue (a disjoin between the kernel API call s_routing, and the userspace call `VIDIOC_S_INPUT`) and this modification will be removed.
+
+Please note that receiving interlaced video is not supported, therefore the ADV7281(A)-M version of the chip is of limited use as it doesn't have the necessary I2P deinterlacing block. Also ensure when selecting a device to specify the -M option. Without that you will get a parallel output bus which can not be interfaced to the Raspberry Pi.
+
+There are no known commercially available boards using these chips, but this driver has been tested via the Analog Devices [EVAL-ADV7282-M evaluation board](https://www.analog.com/en/design-center/evaluation-hardware-and-software/evaluation-boards-kits/EVAL-ADV7282A-M.html)
+
+This driver can be loaded using the `config.txt` dtoverlay entry `adv7282m`, or `adv728x` with a parameter of either `adv7280m=1`,`adv7281m=1`, or `adv7281ma=1` if you are not using the ADV7282-M chip variant. e.g.
+
+```
+dtoverlay=adv728x=adv7280m=1
+```
+
+*Toshiba TC358743 HDMI to CSI2 bridge*
+
+This is a HDM I to CSI-2 bridge chip, capable of converting video data at up to 1080p60. Although capable of 4 lane transfers, on the Raspberry only 2 lanes are used. 
```

Comment:
"HDM I"

Reply:
Drop the second sentence - it's covered later.
(4 lanes can be used on the CM/CM3, therefore it's also ambiguous/wrong)